src/exchanges/OrderFiller.sol

```diff
@@ -0,0 +1,329 @@
+pragma solidity 0.6.1;
+pragma experimental ABIEncoderV2;
+
+import "../dependencies/DSMath.sol";
+import "../dependencies/token/IERC20.sol";
+import "../fund/accounting/IAccounting.sol";
+import "../fund/trading/ITrading.sol";
+
+/// @title Order Filler
+/// @author Melonport AG <team@melonport.com>
+contract OrderFiller is DSMath {
+    event OrderFilled(
+        address indexed exchangeAddress,
+        address targetAsset,
```

Comment:
We discussed a bit in our DMs, but I decided to go with buy/sell for this contract because I wanted it to be agnostic to whether the order is a make or take, especially the event. Marking as resolved, but happy to revisit.